.buildkite/pipeline.yml

```diff
@@ -13,13 +13,23 @@ common_params:
 # This is the default pipeline – it will build and test the app
 steps:
 
+  # Make sure there is a cached version of the pods avaialble before testing
+  # the app or deploying installable builds to avoid parallel steps wasting
+  # time building the same `Podfile.lock` from scratch.
+  - label: ':cocoapods: Pre-load Pods'
+    key: pods
+    command: .buildkite/commands/load-pods.sh
+    env: *common_env
+    plugins: *common_plugins
```

Comment:
Does this really work? I thought that each step in a Buildkite pipeline were independent from one another in terms of the agents they might be using, which means that this step downloading / installing pods might run on one agent, while the following steps that depends on it might run on a different one:

> However you add steps to your pipeline, keep in mind that steps may run on different agents. It is good practice to install your dependencies in the same step that you run them.
> — _Source: https://buildkite.com/docs/pipelines/defining-steps#adding-steps_

So if we want this to work as expected, we'd also have to e.g. upload a `.tar` of the `Pods/` folder that got installed by this step in artefacts, then download and uncompress that `Pods.tar` back in the subsequent steps (similar to how we do it for [`build-for-testing` uploading the Build Products here](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.buildkite/commands/build-for-testing.sh#L19-L21) then [`unit-tests.sh` downloading them back here](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.buildkite/commands/run-unit-tests.sh#L7-L9))

Reply:
> So if we want this to work as expected, we'd also have to e.g...

That's what [`install_cocoapods` does](https://github.com/Automattic/bash-cache-buildkite-plugin/blob/688461e48115844a2e71099b69d4b4e7d7bb6f3e/bin/install_cocoapods) at the `bash-cache` plugin level.

Reply:
Oh I didn't realize that `install_cocoapods` was already I closing the cache dance! Your initial approach makes more sense now 🙂

So I guess the only thing left to solve is how to tell `install_cocoapods` told only install them (and thus ultimately rebuild the cache) if there was a cache miss… but **don't bother installing them during this check step** in case of a cache hit.